runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine.kt

```diff
@@ -179,3 +199,22 @@ private fun toOkHttpTlsVersion(sdkTlsVersion: SdkTlsVersion): OkHttpTlsVersion =
     SdkTlsVersion.TLS_1_2 -> OkHttpTlsVersion.TLS_1_2
     SdkTlsVersion.TLS_1_3 -> OkHttpTlsVersion.TLS_1_3
 }
+
+/**
+ * Creates an SSL context with custom trust and key managers
+ */
+private fun createSslContext(trustManagerProvider: TlsTrustManagersProvider?, keyManagerProvider: TlsKeyManagersProvider?): Pair<SSLContext, X509TrustManager> {
+    val trustManagers = trustManagerProvider?.trustManagers()
+    val keyManagers = keyManagerProvider?.keyManagers()
+
+    if (trustManagerProvider != null && (trustManagers.isNullOrEmpty() || trustManagers[0] !is X509TrustManager)) {
+        throw IllegalStateException("Unexpected trust managers")
```

Comment:
This exception message could be improved. I would recommend breaking out each condition into a separate `check(...) { "useful error message" }` to make it more granular